### PR TITLE
Update health values for battery

### PR DESCRIPTION
**Pull request by bkothapx**
add sysfs read for charge_full and charge_full_design
to update battery health

Tracked-On:
Signed-off-by: Kothapeta, BikshapathiX <bikshapathix.kothapeta@intel.com>

### CODE REVIEW

File: health/2.1/libhealthd/healthd_board_default.cpp [hunk 1]
```diff
@@ -44,6 +45,18 @@ static void parse_battery_status(uint8_t *status)
...
+{
+        if( (mpkt->charge_full == 0) || (mpkt->charge_full_design == 0) ) {
+            s_props.batteryHealth = android::BATTERY_HEALTH_DEAD;
+        }
+        else if ( (((mpkt->charge_full)*100)/mpkt->charge_full_design) >= 80 ) {
+            s_props.batteryHealth = android::BATTERY_HEALTH_GOOD;
+        }
+        else
+            s_props.batteryHealth = android::BATTERY_HEALTH_UNKNOWN;
+}
```
Comment (vilasrk) [inline health/2.1/libhealthd/healthd_board_default.cpp:58]: Indentation is not proper.

**Review by vilasrk — APPROVED** [review]
LGTM

File: health/2.1/libhealthd/healthd_board_default.cpp [hunk 2]
```diff
@@ -85,7 +98,12 @@ static void parse_battery_properties(struct monitor_pkt *mpkt)
     s_props.batteryFullCharge = mpkt->charge_full;
     s_props.batteryChargeCounter = mpkt->charge_now;
     parse_battery_status(mpkt->status);
-    parse_battery_health(mpkt->health);
+    if ( strlen((const char*)mpkt->health) == 0 ){
+	    update_battery_health(mpkt);
+    }
+    else{
```
Comment (JaikrishnaNemallapudi) [inline health/2.1/libhealthd/healthd_board_default.cpp:104]: add space after else {
better do in  one line } else {

File: health/2.1/libhealthd/healthd_board_default.cpp [hunk 3]
```diff
@@ -85,7 +98,12 @@ static void parse_battery_properties(struct monitor_pkt *mpkt)
     s_props.batteryFullCharge = mpkt->charge_full;
     s_props.batteryChargeCounter = mpkt->charge_now;
     parse_battery_status(mpkt->status);
-    parse_battery_health(mpkt->health);
+    if ( strlen((const char*)mpkt->health) == 0 ){
+	    update_battery_health(mpkt);
```
Comment (JaikrishnaNemallapudi) [inline health/2.1/libhealthd/healthd_board_default.cpp:102]: remove extra spaces

File: health/2.1/libhealthd/healthd_board_default.cpp [hunk 4]
```diff
@@ -179,3 +197,9 @@ int healthd_board_battery_update(struct android::BatteryProperties *props)
     }	    
     return 0;
 }
+
+bool get_battery_mediation_present(void)
+{
+    return is_vsock_present;
```
Comment (JaikrishnaNemallapudi) [inline health/2.1/libhealthd/healthd_board_default.cpp:203]: You mean host battery service wont launch without battery ?

File: health/2.1/HealthImpl.cpp [hunk 5]
```diff
@@ -58,9 +59,11 @@ void HealthImpl::UpdateHealthInfo(HealthInfo* health_info) {
     convertFromHealthInfo(health_info->legacy.legacy, &props);
     healthd_board_battery_update(&props);
     convertToHealthInfo(&props, health_info->legacy.legacy);
-    health_info->batteryCapacityLevel = android::hardware::health::V2_1::BatteryCapacityLevel::UNKNOWN;
-    health_info->batteryChargeTimeToFullNowSeconds = 0;
-    health_info->batteryFullChargeDesignCapacityUah = 0;
+    if ( get_battery_mediation_present() == false){
```
Comment (JaikrishnaNemallapudi) [inline health/2.1/HealthImpl.cpp:62]: add space ) {

**Review by JaikrishnaNemallapudi — APPROVED** [review]

**Review by balajim001 — APPROVED** [review]
LGTM

**Pull request merged**